shopinvader_schema_invoice/schemas/invoice.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Camptocamp SA
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date
+
+from extendable_pydantic import StrictExtendableBaseModel
+
+from .amount import InvoiceAmount
+
+
+class Invoice(StrictExtendableBaseModel, extra="ignore"):
+    id: int
+    name: str
+    date_invoice: date
+    date_due: date | None = None
+    state: str
+    payment_state: str
+    ref: str | str
+    payment_reference: str | str
+    amount: InvoiceAmount | None
```

Comment:
Out of curiosity (I'm fine with it) what's your rationale for using a distinct data structure for the amounts?

Reply:
I'm just trying to stay consistent w/ what I've seen so far. Example: https://github.com/shopinvader/odoo-shopinvader/blob/16.0/shopinvader_schema_sale/schemas/amount.py

Reply:
Indeed, thanks.